python/fusion_engine_client/analysis/file_index.py

```diff
@@ -207,7 +210,26 @@ def load(self, index_path, data_path=None, delete_on_error=True):
                     os.remove(index_path)
                 raise ValueError("Index file empty but data file not 0 length. [size=%d B]" % data_file_size)
 
-            # See if the index is larger than the data file.
+            # Get the last entry in the index. If its message type is INVALID, it's a special marker at the end of the
+            # index file indicating the size of the binary data file when the index was created. If it exists, we can
+            # use it to check if the data file size has changed.
```

Comment:
For clarity, what situation is this check detecting?

Reply:
This is looking to see if the size of the `.p1log` file changed at all since the index was originally generated. The most common cause of this is someone re-extracting the contents from an existing mixed-binary file, but hitting Ctrl-C in the middle so the new file is only part of the original file.